### DISCUSSION TITLE
Hotspot STOPS on Android 12 when screen turns off

### DISCUSSION THREAD

**Discussion by hennyhoeven** [Mygod/VPNHotspot#318] [Q&A]
Used it a lot and worked fine to get my laptop online.

Now that I have upgraded that same phone (Mi 10) to android 12, it still works, but...
If the screen of the phone turns of or if you switch to another application on your phone - the hotspot STOPS.

Going back into the hotspot app you have see the toggle for WiFi hotstop back to off..

**Comment by Mygod** [comment]
Try "Keep Wi-Fi alive" option and turn off "turn off hotspot automatically" in your system settings. You can also try to look for other relevant system settings.

**Reply by hennyhoeven** [reply]
Thanks for the quick reply.

I already configured and tried that. 
The different keep alive options and of course disabling 'turn of automatically'.

Still, even when switching to another app and then back to vpn hotspot, the WiFi hotspot is directly turned off.

Maybe interesting to pinpoint the root cause.

If I turn on bluetooth or usb hot spot and then switch apps, those 2 are still enabled.

Only the WiFi hotspots stops on app change or if the screen turns off.

Ps, I have also search all Android/miui power options and configured vpn hotspot to be allowed all battery usage (there are also options to stop background apps).

**Reply by Mygod** [reply]
Which mode are you using? If you were using temporary Wi-Fi hotspot, maybe there is some weird restrictions in MIUI. It is recommended to use system hotspot + this app for best results.
